### ISSUE TITLE
[Question] How to generate heatmaps for visualization using models other than ABMIL?

### ISSUE THREAD

**Issue by xiachashuanghua** [mahmoodlab/TRIDENT#106]
### What's your question?

I am a beginner and have followed the tutorial to generate heatmaps using the ABMIL model. However, when I try other models, they only return a final prediction result without attention maps like ABMIL. How should I modify the code or model to generate heatmaps for these other models?

**Comment by ajv012** [comment]
Thank you for your question. 

To generate a heatmap, the ABMIL model returns both the logits (used for predictions but not for heatmaps) and the post-softmax attention scores, which are the key values needed for heatmap generation (see [here](https://github.com/mahmoodlab/TRIDENT/blob/cb20b652b5e25bbf4e3fb3a810964ec1a9ba3854/trident/slide_encoder_models/model_zoo/reusable_blocks/ABMIL.py#L96)). For context, the attention scores are used to aggregate patch-level features into a slide-level representation.

To enable heatmap creation, you will need to modify your custom model to return these attention scores. Once available, you can use them with our heatmap generation functions.

Let us know if that clarifies your doubts or if you need any additional information.
